netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsServerHandler.java

```diff
@@ -142,7 +142,11 @@ protected void unbufferedWrite(ChannelHandlerContext ctx, HttpStreamsHandler<Htt
                 close = true;
                 continueExpected = false;
             }
-            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)) {
+            // According to RFC 7230 a server MUST NOT send a Content-Length or a Transfer-Encoding when the status
+            // code is 1xx or 204, also a status code 304 may not have a Content-Length or Transfer-Encoding set.
+            int status = out.message.getStatus().code();
+            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)
+                    && !(status < 200 || status == 204 || status == 304)) {
```

Comment:
There are only 3 of the 1xx status codes afaik.

-- 
Cheers,
√
On Mar 24, 2016 7:54 AM, "Schmitt Christian" <notifications@github.com>
wrote:

> In
> netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsServerHandler.java
> <https://github.com/typesafehub/netty-reactive-streams/pull/9#discussion_r57280104>
> :
>
> > @@ -142,7 +142,11 @@ protected void unbufferedWrite(ChannelHandlerContext ctx, HttpStreamsHandler<Htt
> >                  close = true;
> >                  continueExpected = false;
> >              }
> > -            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)) {
> > +            // According to RFC 7230 a server MUST NOT send a Content-Length or a Transfer-Encoding when the status
> > +            // code is 1xx or 204, also a status code 304 may not have a Content-Length or Transfer-Encoding set.
> > +            int status = out.message.getStatus().code();
> > +            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)
> > +                    && !(status < 200 || status == 204 || status == 304)) {
>
> Actually how could I represent the 1xx statuses?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/typesafehub/netty-reactive-streams/pull/9/files/952e09e949215d38e7e7adf522fcf2066020f5b6#r57280104>
>